### PR TITLE
feat(UI): Add status grid widget

### PR DESCRIPTION
**Pull request by Thebarda**
## Description

This implements the status grid widget.

https://github.com/centreon/centreon/assets/12515407/e46c24aa-b913-47ab-a29a-f8b0416e696c

BREAKING CHANGES:
- useFetchQuery/useMutationQuery: the `./api/latest` is no longer required when defining the endpoint. It'll be automatically added by hooks. But if your endpoint does not start with `./api/latest`, add the prop `baseEndpoint` to `true`.

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x
- [x] 24.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/www/install/php/Update-23.10.0.php [hunk 1]
```diff
@@ -356,7 +356,8 @@
                         ('centreon-widget-generictext', '23.10.0'),
                         ('centreon-widget-singlemetric', '23.10.0'),
                         ('centreon-widget-graph', '23.10.0'),
-                        ('centreon-widget-topbottom', '23.10.0')
+                        ('centreon-widget-topbottom', '23.10.0'),
+                        ('centreon-widget-statusgrid', '23.10.0')
```
Comment (a-launois) [inline centreon/www/install/php/Update-23.10.0.php:360]: This won't work if a platform already is in 23.10.0
-> for develop and dev-23.10.x: Update-23.10.2.php or Update-23.10.3.php (depending of release status for 23.10)
-> for develop:  Update-24.10.0.php

Comment (Thebarda) [reply centreon/www/install/php/Update-23.10.0.php:360]: I don't know, the update file is not created

File: centreon/www/install/php/Update-23.10.2.php [hunk 2]
```diff
@@ -123,6 +142,9 @@
 
 try {
     $createDashboardsPlaylistTables($pearDB);
+
+    $populateDahsboardTables($pearDB);
+    
     $dropColumnVersionFromDashboardWidgetsTable($pearDB);
 } catch (\Exception $e) {
```
Comment (jeremyjaouen) [inline centreon/www/install/php/Update-23.10.2.php:150]:
```suggestion
    $dropColumnVersionFromDashboardWidgetsTable($pearDB);
 
    if (! $pearDB->inTransaction()) {
        $pearDB->beginTransaction();
    }
    $populateDahsboardTables($pearDB);
    
    $pearDB->commit();
} catch (\Exception $e) {
    if ($pearDB->inTransaction()) {
        $pearDB->rollBack();
    }
```

Comment (jeremyjaouen) [reply centreon/www/install/php/Update-23.10.2.php:150]: The drop column should be before transaction, and populate into transaction

**Review by jeremyjaouen — CHANGES_REQUESTED** [review]

**Review by jeremyjaouen — APPROVED** [review]
Ok for upgrade part

File: centreon/www/install/php/Update-24.04.0.php [hunk 3]
```diff
@@ -211,6 +211,25 @@
     }
 };
 
+$populateDahsboardTables = function(CentreonDb $pearDB): void {
```
Comment (callapa) [inline centreon/www/install/php/Update-24.04.0.php:214]: Unnecessary update file.
If the user comes from a version lower than 23.02, the update-23.10.2 file contains all the code needed to update the database.
From a new installation, the modification will be made with the inserBaseConf.sql file.

File: centreon/www/install/php/Update-23.10.2.php [hunk 4]
```diff
@@ -110,6 +110,25 @@
     );
 };
 
+$populateDahsboardTables = function(CentreonDb $pearDB): void {
+    if ($pearDB->isColumnExist('dashboard_widgets', 'name')) {
```
Comment (callapa) [inline centreon/www/install/php/Update-23.10.2.php:114]: This instruction is used to find out if a column exists before creating it.
This is not the case here.

File: centreon/www/install/php/Update-23.10.2.php [hunk 5]
```diff
@@ -122,10 +139,20 @@
...
-} catch (\Exception $e) {
+    
+    $createDashboardsPlaylistTables($pearDB);
 
+    if (! $pearDB->inTransaction()) {
+        $pearDB->beginTransaction();
+    }
+    $populateDahsboardTables($pearDB);
+    
+    $pearDB->commit();
```
Comment (kduret) [inline centreon/www/install/php/Update-23.10.2.php:151]:
```suggestion
    if ($pearDB->inTransaction()) {
        $pearDB->commit();
    }
```

**Review by callapa — APPROVED** [review]

**Review by sc979 — DISMISSED** [review]
LGTM on action.yml

**Review by Med1233 — DISMISSED** [review]

File: centreon/lang/es_ES.UTF-8/LC_MESSAGES/messages.po [hunk 6]
```diff
@@ -16584,6 +16584,54 @@ msgstr "Modo avanzado"
...
+# msgid "Unselect all"
+# msgstr ""
+
+# msgid "Display up to"
+# msgstr ""
+
+# msgid "tiles"
+# msgstr ""
+
+# msgid "Waiting for fullfilled resources."
```
Comment (smau-centreon) [inline centreon/lang/es_ES.UTF-8/LC_MESSAGES/messages.po:16596]:
```suggestion
# msgid "You must fill in more resources."
```

File: centreon/lang/es_ES.UTF-8/LC_MESSAGES/messages.po [hunk 7]
```diff
@@ -16584,6 +16584,54 @@ msgstr "Modo avanzado"
...
+# msgid "For a service, the service status is displayed."
+# msgstr ""
+
+# msgid "Display resources with this state"
+# msgstr ""
+
+# msgid "Sort by"
+# msgstr ""
+
+# msgid "See more on the Resource status page"
```
Comment (smau-centreon) [inline centreon/lang/es_ES.UTF-8/LC_MESSAGES/messages.po:16632]:
```suggestion
# msgid "See more on the Resources Status page"
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 8]
```diff
@@ -18345,6 +18345,54 @@ msgstr "Gestion de l'accès aux ressources"
...
+msgid "Unselect all"
+msgstr "Tout désélectionner"
+
+msgid "Display up to"
+msgstr "Afficher jusqu'à"
+
+msgid "tiles"
+msgstr "tuiles"
+
+msgid "Waiting for fullfilled resources."
```
Comment (smau-centreon) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:18357]:
```suggestion
msgid "You must fill in more resources."
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 9]
```diff
@@ -18345,6 +18345,54 @@ msgstr "Gestion de l'accès aux ressources"
...
+msgstr "En attente de ressources remplies."
```
Comment (smau-centreon) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:18358]:
```suggestion
msgstr "Vous devez indiquer plus de ressources."
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 10]
```diff
@@ -18345,6 +18345,54 @@ msgstr "Gestion de l'accès aux ressources"
...
+msgstr "Grille de statut"
+
+msgid "Displays a status overview for selected resources."
+msgstr "Affiche une vue d'ensemble du statut des ressources sélectionnées."
+
+msgid "Display with this resource type"
+msgstr "Afficher avec ce type de ressource"
+
+msgid "Display resources with this status"
+msgstr "Afficher les ressources avec ce status"
```
Comment (smau-centreon) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:18379]:
```suggestion
msgstr "Afficher les ressources avec ce statut"
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 11]
```diff
@@ -18345,6 +18345,54 @@ msgstr "Gestion de l'accès aux ressources"
...
+msgid "For a service, the service status is displayed."
+msgstr "Pour un service, le statut du service est affiché."
+
+msgid "Display resources with this state"
+msgstr "Afficher les ressources avec cet état"
+
+msgid "Sort by"
+msgstr "Trier par"
+
+msgid "See more on the Resource status page"
```
Comment (smau-centreon) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:18393]:
```suggestion
msgid "See more on the Resources status page"
```

File: centreon/lang/pt_BR.UTF-8/LC_MESSAGES/messages.po [hunk 12]
```diff
@@ -17048,6 +17048,54 @@ msgstr ""
...
+# msgid "Unselect all"
+# msgstr ""
+
+# msgid "Display up to"
+# msgstr ""
+
+# msgid "tiles"
+# msgstr ""
+
+# msgid "Waiting for fullfilled resources."
```
Comment (smau-centreon) [inline centreon/lang/pt_BR.UTF-8/LC_MESSAGES/messages.po:17060]:
```suggestion
# msgid "You must fill in more resources."
```

File: centreon/lang/pt_BR.UTF-8/LC_MESSAGES/messages.po [hunk 13]
```diff
@@ -17048,6 +17048,54 @@ msgstr ""
...
+# msgid "For a service, the service status is displayed."
+# msgstr ""
+
+# msgid "Display resources with this state"
+# msgstr ""
+
+# msgid "Sort by"
+# msgstr ""
+
+# msgid "See more on the Resource status page"
```
Comment (smau-centreon) [inline centreon/lang/pt_BR.UTF-8/LC_MESSAGES/messages.po:17096]:
```suggestion
# msgid "See more on the Resources status page"
```

File: centreon/lang/pt_PT.UTF-8/LC_MESSAGES/messages.po [hunk 14]
```diff
@@ -17037,6 +17037,54 @@ msgstr "Você não salvou as alterações, deseja continuar?"
...
+# msgid "Unselect all"
+# msgstr ""
+
+# msgid "Display up to"
+# msgstr ""
+
+# msgid "tiles"
+# msgstr ""
+
+# msgid "Waiting for fullfilled resources."
```
Comment (smau-centreon) [inline centreon/lang/pt_PT.UTF-8/LC_MESSAGES/messages.po:17049]:
```suggestion
# msgid "You must fill in more resources."
```

File: centreon/lang/pt_PT.UTF-8/LC_MESSAGES/messages.po [hunk 15]
```diff
@@ -17037,6 +17037,54 @@ msgstr "Você não salvou as alterações, deseja continuar?"
...
+# msgid "For a service, the service status is displayed."
+# msgstr ""
+
+# msgid "Display resources with this state"
+# msgstr ""
+
+# msgid "Sort by"
+# msgstr ""
+
+# msgid "See more on the Resource status page"
```
Comment (smau-centreon) [inline centreon/lang/pt_PT.UTF-8/LC_MESSAGES/messages.po:17085]:
```suggestion
# msgid "See more on the Resources status page"
```

File: centreon/www/widgets/src/centreon-widget-statusgrid/src/translatedLabels.ts [hunk 16]
```diff
@@ -0,0 +1,11 @@
+export const labelNoResources = 'Waiting for fullfilled resources.';
```
Comment (smau-centreon) [inline centreon/www/widgets/src/centreon-widget-statusgrid/src/translatedLabels.ts:1]:
```suggestion
export const labelNoResources = 'You must fill in more resources.';

```

File: centreon/www/widgets/src/centreon-widget-statusgrid/src/translatedLabels.ts [hunk 17]
```diff
@@ -0,0 +1,11 @@
...
+export const labelServiceName = 'Service name';
+export const labelAllServicesAreWorkingFine =
+  'All services on this host are working fine.';
+export const labelAcknowledged = 'Acknowledged';
+export const labelInDowntime = 'In downtime';
+export const labelAllMetricsAreWorkingFine =
+  'All metrics on this service are working fine.';
+export const labelMetricName = 'Metric name';
+export const labelValue = 'Value';
+export const labelSeeMore = 'See more on the Resource status page';
```
Comment (smau-centreon) [inline centreon/www/widgets/src/centreon-widget-statusgrid/src/translatedLabels.ts:11]:
```suggestion
export const labelSeeMore = 'See more on the Resources status page';

```

File: centreon/lang/de_DE.UTF-8/LC_MESSAGES/messages.po [hunk 18]
```diff
@@ -16251,6 +16251,54 @@ msgstr "Erweiterter Modus"
...
+# msgid "Unselect all"
+# msgstr ""
+
+# msgid "Display up to"
+# msgstr ""
+
+# msgid "tiles"
+# msgstr ""
+
+# msgid "Waiting for fullfilled resources."
```
Comment (smau-centreon) [inline centreon/lang/de_DE.UTF-8/LC_MESSAGES/messages.po:16263]:
```suggestion
# msgid "You must fill in more resources."
```

**Review by smau-centreon — CHANGES_REQUESTED** [review]

**Review by a-launois — DISMISSED** [review]

**Review by Med1233 — APPROVED** [review]

**Review by jeremyjaouen — APPROVED** [review]

File: centreon/lang/pt_BR.UTF-8/LC_MESSAGES/messages.po [hunk 19]
```diff
@@ -17048,6 +17048,54 @@ msgstr ""
...
+# msgid "For a service, the service status is displayed."
+# msgstr ""
+
+# msgid "Display resources with this state"
+# msgstr ""
+
+# msgid "Sort by"
+# msgstr ""
+
+# msgid "See more on the Resources status page"
```
Comment (smau-centreon) [inline centreon/lang/pt_BR.UTF-8/LC_MESSAGES/messages.po:17096]:
```suggestion
# msgid "See more on the Resources Status page"
```

File: centreon/lang/pt_PT.UTF-8/LC_MESSAGES/messages.po [hunk 20]
```diff
@@ -17037,6 +17037,54 @@ msgstr "Você não salvou as alterações, deseja continuar?"
...
+# msgid "For a service, the service status is displayed."
+# msgstr ""
+
+# msgid "Display resources with this state"
+# msgstr ""
+
+# msgid "Sort by"
+# msgstr ""
+
+# msgid "See more on the Resources status page"
```
Comment (smau-centreon) [inline centreon/lang/pt_PT.UTF-8/LC_MESSAGES/messages.po:17085]:
```suggestion
# msgid "See more on the Resources Status page"
```

File: centreon/www/widgets/src/centreon-widget-statusgrid/src/translatedLabels.ts [hunk 21]
```diff
@@ -0,0 +1,12 @@
...
+export const labelServiceName = 'Service name';
+export const labelAllServicesAreWorkingFine =
+  'All services on this host are working fine.';
+export const labelAcknowledged = 'Acknowledged';
+export const labelInDowntime = 'In downtime';
+export const labelAllMetricsAreWorkingFine =
+  'All metrics on this service are working fine.';
+export const labelMetricName = 'Metric name';
+export const labelValue = 'Value';
+export const labelSeeMore = 'See more on the Resources status page';
```
Comment (smau-centreon) [inline centreon/www/widgets/src/centreon-widget-statusgrid/src/translatedLabels.ts:12]:
```suggestion
export const labelSeeMore = 'See more on the Resources Status page';

```

**Review by Med1233 — DISMISSED** [review]

**Review by Med1233 — DISMISSED** [review]

File: centreon/www/install/php/Update-23.10.2.php [hunk 22]
```diff
@@ -110,6 +110,23 @@
     );
 };
 
+$populateDashboardTables = function(CentreonDb $pearDB): void {
```
Comment (jeremyjaouen) [inline centreon/www/install/php/Update-23.10.2.php:113]: This should be move in 23.10.3 script, as the 23.10.2 is currently being released

Comment (Thebarda) [reply centreon/www/install/php/Update-23.10.2.php:113]: :')

File: centreon/www/install/php/Update-23.10.2.php [hunk 23]
```diff
@@ -49,9 +49,24 @@
 };
 
 try {
+    $dropColumnVersionFromDashboardWidgetsTable($pearDB);
+
```
Comment (jeremyjaouen) [inline centreon/www/install/php/Update-23.10.2.php:53]:
```suggestion
```
This has been moved to 22.10.3 thats why your e2e tests failed

"Value of type null is not callable"

**Review by jeremyjaouen — CHANGES_REQUESTED** [review]

**Review by Med1233 — APPROVED** [review]

**Pull request merged**